cmd/argo/commands/common/wait.go

```diff
@@ -69,7 +69,16 @@ func waitOnOne(serviceClient workflowpkg.WorkflowServiceClient, ctx context.Cont
 			continue
 		}
 		wf := event.Object
-		if wf != nil && !wf.Status.FinishedAt.IsZero() {
+
+		isWorkflowInTerminalState := func(wf *wfv1.Workflow) bool {
+			if !wf.Status.FinishedAt.IsZero() {
+				return true
+			} else {
+				return wf.Status.Phase == wfv1.WorkflowFailed || wf.Status.Phase == wfv1.WorkflowError || wf.Status.Phase == wfv1.WorkflowSucceeded
```

Comment:
This is available as `wf..Status.Phase.Completed()`

Reply:
Thanks for the suggestion! I switched the condition to use it